assets/agenda/components/AgendaList.tsx

```diff
@@ -300,12 +300,60 @@ class AgendaList extends React.Component<IProps, IState> {
         const isExtended = this.props.activeView === EXTENDED_VIEW;
         const showShortcutActionIcons = shouldShowListShortcutActionIcons(this.props.listConfig, isExtended);
 
+        const getItemIdsSorted = () => {
+            let topStoryIds: Array<string> = [];
+            let coveragesOnlyIds: Array<string> = [];
+            let restIds: Array<string> = [];
+
+            itemIds.forEach((id) => {
+                const item = this.props.itemsById[id];
+                const hasCoverage = (item.coverages?.length ?? 0) > 0;
+                const isTopStory = (item.subject ?? []).find((subj) => subj.scheme === this.props.listConfig.subject?.topStoryScheme);
+
+                if (isTopStory) {
+
+                    // hiddenItems has items which are multiDay and are not on the first date of the group
+                    if (group.hiddenItems.includes(item._id) === false) {
+                        topStoryIds = [
```

Comment:
We're not dealing with a redux store here, so using an immutable approach is an overkill here. I'd use `Array.push` instead.